055-ChaosStudio4AKS/Student/Challenge-04.md

```diff
@@ -0,0 +1,33 @@
+# Challenge 04 - [Optional] Injecting Chaos into your CI/CD pipeline
+
+[< Previous Challenge](./Challenge-03.md) - **[Home](../README.md)**
+
+## Pre-requisites
+To complete this challenge you will using the Pizza Application or your AKS application
+
+
+## Introduction
+You will need to have an indepth understanding of DevOps and your CI/CD tool of choice.
+
+This is where the rubber meets the road. You will take what you have learned from the previous challenges and apply the knowledge here. 
+
+
+## Description
+In this challenge you will conduct a chaos experiment in your CI/CD pipeline.
+You will take the Pizza Application or your application and add a chaos experiment to your deployment pipeline.
+Run your experiments in Dev/Test, do not run in Prod.
+
+
+## Tips
+1. You want your application to be available (healthy state) during failure.
+2. What kinds of faults and remediations come to mind from the previous challenges? 
+
+## Success Criteria
+
+- Chaos Studio injects fault(s) into your application.
```

Comment:
We should update the success criteria here to include some verbiage where it includes implementing this in a CI/CD pipeline.

Also what is the expectation here for the faults?  Do they just implement the fault in the pipeline?  Or are some type of tests also in the pipeline to verify the app is healthy?

Reply:
Maybe you and I should work on this one. I have instructions for a simple VM test versus rigging up the Pizza App in a CI/CD pipeline